platform-sdk/swirlds-state-api/src/main/java/com/swirlds/state/spi/ReadableKVStateBase.java

```diff
@@ -40,9 +40,7 @@ public abstract class ReadableKVStateBase<K, V> implements ReadableKVState<K, V>
      * changed before we got to handle transaction. If the value is "null", this means it was NOT
      * FOUND when we looked it up.
      */
-    private final ConcurrentMap<K, V> readCache = new ConcurrentHashMap<>();
-
-    private final Set<K> unmodifiableReadKeys = Collections.unmodifiableSet(readCache.keySet());
+    private ConcurrentMap<K, V> readCache = new ConcurrentHashMap<>();
```

Comment:
Should revert back to final, change to Map interface, and populate default only in one-arg constructor.

```suggestion
    private final Map<K, V> readCache;
```

Reply:
I see that it's a concurrent map on `main`: https://github.com/hiero-ledger/hiero-consensus-node/blob/main/platform-sdk/swirlds-state-api/src/main/java/com/swirlds/state/spi/ReadableKVStateBase.java#L43 .

I'd guess this was done for a reason, probably to indicate that this map is in fact used concurrently, which is extremely important. A regular map (e.g. a HashMap) may simply not work here.

So I suggest to keep using the ConcurrentMap here.

Reply:
But yes, it should be `final` still. We want to avoid using mutable fields as much as possible because they're error-prone in multi-threaded systems.

Reply:
For us, it doesn't need to be concurrent since we're using virtual threads and `ScopedValue` to store the state. But probably we can use `ForwardingConcurrentMap` just as easily as we were intending to use `ForwardingMap`.

Reply:
Well, this `ReadableKVStateBase` class is a part of the consensus node, and it does seem to require this map to be concurrent. Your particular usage of this class may not trigger concurrent operations, but in general, this class does support them, and hence it should require a `ConcurrentMap` as a cache.